### DISCUSSION TITLE
Accept cast `object`

### DISCUSSION THREAD

**Discussion by ghost** [filamentphp/filament#8218] [Ideas]
I dont know if it is easy to add this functionality, but when using the following:

```php
\Filament\Forms\Components\TextInput::make('address.street')
```

The `address` property must be cast using either `'json'` (or `'array'` which is the same). Casting as `'object'` does not fill in the text input (it stays blank). 

I prefer casting to object because it has a neater syntax which is `$record->address->street` vs `$record->address['street']`.

**Comment by danharrin** [comment]
It is difficult as I don't think Livewire supports nested stdObject inside public property arrays

**Comment by juancamunozfp** [comment]
@danharrin is there still no way to fix this??

**Reply by danharrin** [reply]
I don't know, never tried it.
